Tests/CometTests/Authenticator/AuthenticatorTests.swift

```diff
@@ -0,0 +1,436 @@
+//
+//  AuthenticatorTests.swift
+//  Comet
+//
+//  Created by Tuan Tu Do on 26.05.2021.
+//  Copyright © 2021 Etnetera. All rights reserved.
+//
+
+import Combine
+@testable import Comet
+import XCTest
+
```

Comment:
Consider using a Combine testing framework like [combine-schedulers](https://github.com/pointfreeco/combine-schedulers) or [Entwine](https://github.com/tcldr/Entwine). Both libraries provide a test scheduler. Entwine has a better testing API, however combine-schedulers has a type-erased scheduler (AnyScheduler) that helps you avoid generics for schedulers.

Both will help you run your tests instantaneously, instead of waiting for the 2/3 seconds.

Reply:
I don't want any 3rd party dependencies in a small project like this one. I guess I am OK with waiting several seconds in exchange of not having any unnecessary dependencies.

Reply:
I'm hoping Apple announces something at WWDC. But in this case the dependency would be limited to the tests target. Anyhow, you can always write your own test scheduler...

Reply:
I guess a lot of developers have high hopes this year. We are in times when people start to use `Combine` and `SwiftUI` more in production. Anyway, I think I am fine with the solution that we have right now.

Reply:
GitHub Actions are for free for open source projects LOL